### PR TITLE
clean up exchange configuration handling

### PR DESCRIPTION
**Pull request by arnetheduck**
Per spec, we should not be sending our detected terminal block to EL - the EL configuration exchange should only look at values from configuration and report mismatches.

* note TODO about usage of terminal block hash configuration

### CODE REVIEW

File: beacon_chain/eth1/eth1_monitor.nim [hunk 1]
```diff
@@ -561,67 +559,46 @@ proc exchangeTransitionConfiguration*(p: Eth1Monitor): Future[EtcStatus] {.async
   if dataProvider.isNil:
     return EtcStatus.exchangeError
 
+  # https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#engine_exchangetransitionconfigurationv1
```
Comment (tersec) [inline beacon_chain/eth1/eth1_monitor.nim]: https://github.com/ethereum/execution-apis/blob/v1.0.0-beta.1/src/engine/specification.md#engine_exchangetransitionconfigurationv1

Comment (arnetheduck) [inline beacon_chain/eth1/eth1_monitor.nim]:
```suggestion
  # https://github.com/ethereum/execution-apis/blob/v1.0.0-beta.1/src/engine/specification.md#engine_exchangetransitionconfigurationv1
```

File: beacon_chain/spec/presets.nim [hunk 2]
```diff
@@ -38,8 +38,7 @@ type
 
     # Transition
     TERMINAL_TOTAL_DIFFICULTY*: UInt256
-    TERMINAL_BLOCK_HASH*: BlockHash
-    # TODO TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH*: Epoch
+    TERMINAL_BLOCK_HASH*: BlockHash # TODO use in eht1monitor
```
Comment (tersec) [inline beacon_chain/spec/presets.nim:41]:
```suggestion
    TERMINAL_BLOCK_HASH*: BlockHash # TODO use in eth1monitor
```

**Review by tersec — APPROVED** [review]

File: beacon_chain/eth1/eth1_monitor.nim [hunk 3]
```diff
@@ -561,67 +559,46 @@ proc exchangeTransitionConfiguration*(p: Eth1Monitor): Future[EtcStatus] {.async
...
-      if p.terminalBlockHash.isSome:
-        p.terminalBlockHash.get
-      else:
-        (static default BlockHash),
-    terminalBlockNumber:
-      if p.terminalBlockNumber.isSome:
-        p.terminalBlockNumber.get
-      else:
-        (static default Quantity))
+    terminalBlockHash: p.depositsChain.cfg.TERMINAL_BLOCK_HASH,
```
Comment (zah) [inline beacon_chain/eth1/eth1_monitor.nim:565]: The old code was not fully wrong here I think. We still need to consider the command-line override that the user may supply.

Comment (arnetheduck) [reply beacon_chain/eth1/eth1_monitor.nim:565]: we consider the command line here: https://github.com/status-im/nimbus-eth2/blob/667c3c97ebe3c40fd724cf80c991410e1cd32900/beacon_chain/nimbus_beacon_node.nim#L1830

**Pull request merged**